contracts/CrossChain/Strategy2Crv.sol

```diff
@@ -0,0 +1,211 @@
+        IERC20(0x8D11eC38a3EB5E956B052f67Da8Bdc9bef8Abf3E);
+    IERC20 public constant wftm =
+        IERC20(0x21be370D5312f44cB42ce377BC9b8a0cEF1A4C83);
+
+    address public constant voter = 0x72a34AbafAB09b15E7191822A679f28E067C4a16; // sms
+
+    // rewards token info. we can have more than 1 reward token but this is rare, so we don't include this in the template
+    IERC20 public rewardsToken;
+    bool public hasRewards;
+
+    // Target trades
+    address public rewardsTarget;
+    address public crvTarget;
+
+    /* ========== CONSTRUCTOR ========== */
+
+    constructor(
+        address _vault,
+        address _tradeFactory,
+        address _curvePool,
+        address _gauge,
+        bool _hasRewards,
+        address _rewardsToken,
+        string memory _name
+    ) public StrategyCurveBaseUsingGauge(_vault, _tradeFactory) {
+          // You can set these parameters on deployment to whatever you want
+          maxReportDelay = 7 days; // 7 days in seconds
+          debtThreshold = 5 * 1e18; // we shouldn't ever have debt, but set a bit of a buffer
+          profitFactor = 10_000; // in this strategy, profitFactor is only used for telling keep3rs when to move funds from vault to strategy
+          healthCheck = 0xf13Cd6887C62B5beC145e30c38c4938c5E627fe0; // health.ychad.eth
+
+          // set our keepCRV
+          // keepCRV = 0; not needed since we set it to 0
+
+          // setup our rewards if we have them
+          if (_hasRewards) {
+              hasRewards = true;
+              rewardsToken = IERC20(_rewardsToken);
+              rewardsTarget = address(dai);
+          }
+
+          // this is the pool specific to this vault, but we only use it as an address
+          curve = address(_curvePool);
+
+          // set our curve gauge contract
+          gauge = IGauge(_gauge);
+
+          // set our strategy's name
+          stratName = _name;
+
+          // these are our approvals and path specific to this contract
+          dai.approve(address(curve), type(uint256).max);
+          usdc.approve(address(curve), type(uint256).max);
+          want.approve(address(gauge), type(uint256).max);
+
+          // start off with dai
+          crvTarget = address(dai);
+    }
+
+
+
+    /* ========== MUTATIVE FUNCTIONS ========== */
+    // these will likely change across different wants.
+
+    function prepareReturn(uint256 _debtOutstanding)
+        internal
+        override
+        returns (
+            uint256 _profit,
+            uint256 _loss,
+            uint256 _debtPayment
+        )
+    {
+        // if we have anything in the gauge, then harvest CRV from the gauge
+        uint256 _stakedBal = stakedBalance();
+        if (_stakedBal > 0) {
+            gauge.claim_rewards();
+            uint256 _crvBalance = crv.balanceOf(address(this));
+            // if we claimed any CRV, then sell it
+            if (_crvBalance > 0) {
+                // keep some of our CRV to increase our boost
```

Comment:
Is the idea that SMS would send this CRV cross-chain and lock it on mainnet?

Reply:
yes. Made the default 0.